src/src/Form/CustomerSyncFormDataConfiguration.php

```diff
@@ -0,0 +1,137 @@
+
+    public function __construct(ConfigurationInterface $configuration)
+    {
+        $this->configuration = $configuration;
+    }
+
+    public function getConfiguration(): array
+    {
+        $return = [];
+
+        $return['enabled'] = $this->configuration->getBoolean('SMAILY_ENABLE_CUSTOMER_SYNC');
+        $return['sync_additional'] = json_decode($this->configuration->get('SMAILY_SYNCRONIZE_ADDITIONAL'), true);
+        $return['cron_token'] = $this->configuration->get('SMAILY_CUSTOMER_CRON_TOKEN');
+        $return['cron_url'] = $this->buildCronURL($return['cron_token']);
+        $return['optin_enabled'] = $this->configuration->getBoolean('SMAILY_OPTIN_ENABLED');
+        $return['autoresponder'] = $this->configuration->get('SMAILY_OPTIN_AUTORESPONDER');
+
+        return $return;
+    }
+
+    public function updateConfiguration(array $formData): array
+    {
+        $errors = [];
+
+        $syncEnabled = $formData['enabled'];
+        $additionalFields = $formData['sync_additional'];
+        $optInEnabled = $formData['optin_enabled'];
+        $cronToken = $formData['cron_token'];
+        $autoresponder = $formData['autoresponder'];
+
+        if (empty($cronToken)) {
+            $errors[] = [
+                'key' => 'Please provide a cron token for customer synchronization.',
+                'domain' => 'Modules.Smailyforprestashop.Admin',
+                'parameters' => [],
+            ];
+        }
+
+        if ($optInEnabled && empty($autoresponder)) {
+            $errors[] = [
+                'key' => 'Please select an automation workflow for customer Opt-In trigger.',
+                'domain' => 'Modules.Smailyforprestashop.Admin',
+                'parameters' => [],
+            ];
+        }
+
+        // Clear autoresponder when disabling customer sign-up;
+        if (!$optInEnabled) {
+            $autoresponder = null;
+        }
+
+        if ($this->validateConfiguration($formData) && empty($errors)) {
+            $this->configuration->set('SMAILY_ENABLE_CUSTOMER_SYNC', $syncEnabled);
+            $this->configuration->set('SMAILY_SYNCRONIZE_ADDITIONAL', json_encode($additionalFields));
+            $this->configuration->set('SMAILY_CUSTOMER_CRON_TOKEN', $cronToken);
+            $this->configuration->set('SMAILY_OPTIN_ENABLED', $optInEnabled);
+            $this->configuration->set('SMAILY_OPTIN_AUTORESPONDER', $autoresponder);
+
+            if ($optInEnabled) {
+                // We disable ps_emailsubscription plugin mail sending in order provide a place
+                // for sending opt-in emails and to avoid sending duplicate emails to customers.
+                // These emails can and should be implemented as a part of the automation workflow.
+                $this->configuration->set('NW_VERIFICATION_EMAIL', 0);
+                $this->configuration->set('NW_CONFIRMATION_EMAIL', 0);
+                $this->configuration->set('NW_VOUCHER_CODE', null);
+            }
```

Comment:
Should email sending be restored when opt-in feature is disabled? Or where can the user go and re-enable email sending?

Reply:
Under the `Theme modules` section, there is a PrestaShop built-in module called `Newsletter subscriptions`. Users can re-enable these settings there.

![image](https://github.com/sendsmaily/smaily-prestashop-module/assets/22918359/f31cda93-1632-4e7b-8f18-52f563c1e520)

Reply:
I think this should be described somewhere. The best place would be next to enabling the newsletter subscription form's Smaily integration. A link to the `Theme modules`'s `Newsletter subscriptions` settings would be ideal.

Reply:
I have gone over a few options to better notify users about the `Smaily` and `Newsletter Subscription` module connection.

I have added a section with a module link to the help text. Ideally, the previous help text could be a tooltip for the label. This would shorten the help section and allow better focus on the text but after trying to implement it I ran into multiple bugs. The current tooltip solution is kind of hacky and hard to maintain so I dropped this.
![image](https://github.com/sendsmaily/smaily-prestashop-module/assets/22918359/625d5c77-01fd-4240-8e6c-8b74ec4a2765)

I have also changed the notice type to a warning and planted a module link inside the help text. This will make the message more prominent and noticeable.
![image](https://github.com/sendsmaily/smaily-prestashop-module/assets/22918359/ad4b9d53-dc54-4b86-9da4-1bd88054e322)

https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/818b2033b591912609b55a85ae5dd2cc978dd198

Reply:
The text in the warning message is slightly misleading, because automation workflow doesn't need to be selected for the module to disable "PS Newsletter Subscription" module's email sending.

Would this make the wording align better with the functionality?

> You have enabled triggering opt-in on customer sign-up. We have disabled [...]

Reply:
True, I have improved the wording. https://github.com/sendsmaily/smaily-prestashop-module/pull/57/commits/53a43c7c5a9f94cfd8720c8dab02ab311b233d8a